### DISCUSSION TITLE
Diffenence with clash-verge-rev? 和 clash-verge-rev 的差异？

### DISCUSSION THREAD

**Discussion by Fungus-00** [libnyanpasu/clash-nyanpasu#1003] [General]
**I shall delete this discussion if there is any offence.**

This project seems quite similar to [clash verge rev](https://github.com/clash-verge-rev/clash-verge-rev) as they were forked from clash verge nearly at the same time. I haven't tried the latter though, really curious about their difference. Can anyone having used both tell the distinction?

---

**如有任何冒犯，我将删除此贴**

这个项目看起来和 [clash verge rev](https://github.com/clash-verge-rev/clash-verge-rev) 神似，或许是因为它们在几乎同一时期从 clash verge 分叉出来。尽管我没试过后者，确实很好奇它们的差别。有任何使用过这两款的人说说有何不同之处吗？

**Comment by keiko233** [comment]
简单梳理一下吧

Nyanpasu 是从原 verge 1.3.7 fork 而来，并继承其所有特性；而 verge rev 是从原 verge 1.3.8 fork 并合并 Nyanpasu 1.4.0的修改而来。

Nyanpasu 从1.4开始引入大量独有特性；从1.5开始与 verge 做大差异化开发，添加更多特性；Nyanpasu 1.6将在前端做重构，整体设计靠拢 Google Material You。

Nyanpasu 有多内核兼容，由于内核间会有差异，为了抹平这一问题，Nyanpasu 优先使用自有实现功能。

如果有其他的疑问可以先参考下面的链接查阅。

关于名字和图标 https://nyanpasu.elaina.moe/introduction#story-behind-the-name-and-icon

nyanpasu的起源 https://nyanpasu.elaina.moe/introduction#origin-of-clash-nyanpasu

个人的阐述 https://nyanpasu.elaina.moe/others/issues#what-is-the-significance-of-this-project

**Comment by greenhat616** [comment]
补充一点，nyanpasu 一些特性，譬如内核的更新，会出于抹平内核差异的目的，在 GUI 这侧进行实现，而 verge rev 只兼容 Meta，因此会直接采用 Meta 自带的特性直接暴露。

**Comment by test4uninstall** [comment]
以下是我的主观看法，仅使用了两者的Windows版本。
1.Clash Verge Rev仅支持Meta（mihomo）内核，而Clash Nyanpasu不仅支持Meta（mihomo）内核，还支持传统内核和Rust内核。
2.Clash Verge Rev支持单独设置Http端口和Socks5端口，而Clash Nyanpasu仅支持设置混合端口。
3.Clash Verge Rev不支持Clash字段设置和应用目录设置，而Clash Nyanpasu支持Clash字段设置和应用目录设置。
4.Clash Verge Rev的UI有点冷的感觉，有圆角但偏方形，整体感觉更清晰，而Clash Nyanpasu的UI圆角幅度稍大，整体感觉更可爱，两者的共同点就是简洁。

**Comment by AndrewGoal** [comment]
对于url-test类型的代理分组，clash-verge-rev 支持用户点选干预（这可能是meta内核的特性，也是我需要的方便功能），而本尊不支持💁

**Reply by keiko233** [reply]
我没理解错的话就是verge的在proxies上的工具栏修改test ur，这个功能不准备纳入nyanpasu中，出于一下几点考虑

1. proxy group本身就支持指定 test url https://wiki.metacubex.one/config/proxy-groups/#url ，这个功能支持已经提交到主线
2. nyanpasu 提供了script，如果需要客制化修改可以使用proxy chain进行修改
3. 直接对配置文件修改的功能也是有的

上述几个方法都比直接在proxies修改来的更好

**Comment by SinetianLiu** [comment]
讲真的，我更喜欢Nyanpasu就是出于它的UI设计……Clash Verge Rev功能性太强了。
